### PR TITLE
[BUGFIX release] fix issue with class not recomputing

### LINKED ISSUE #16505: [3.1.0] "class" attribute not being recomputed
We have something like this in a template:

```hbs
{{some-component classNames="some-class fade" class=(if isLoggedIn "in")}}
```

This used to work fine until `3.1.0` - that is, when `isLoggedIn` is true, in addition to the classes in `classNames`, `in` is added. When `isLoggedIn` is false, then only the classes from `classNames` are used. However, in `3.1.0` this doesn't work any more. `in` is not added even when `isLoggedIn` is true. It works only if we add a non-empty string to the else clause - `(if isLoggedIn "in" "some-non-relevant-class")`.

### PR DESCRIPTION
**Pull request by bekzod**
fixes #16505

### CODE REVIEW

File: packages/ember-glimmer/tests/integration/components/curly-components-test.js [hunk 1]
```diff
@@ -404,9 +404,9 @@ moduleFor(
     ['@test class property on components can be dynamic']() {
       this.registerComponent('foo-bar', { template: 'hello' });
 
-      this.render('{{foo-bar class=(if fooBar "foo-bar")}}', {
-        fooBar: true,
-      });
+      this.render('{{foo-bar class=(if fooBar "foo-bar")}}');
+
+      this.runTask(() => set(this.context, 'fooBar', true));
```
Comment (bekzod) [inline packages/ember-glimmer/tests/integration/components/curly-components-test.js]: this would break test before

**Comment by rwjblue** [comment]
Thank you for fixing! Can you grab the tests from this commit by @ef4 (and reset this particular test)?

https://github.com/emberjs/ember.js/commit/5f7e7294bf111d1cb82237d6c8deacffc4e4e364

**Comment by rwjblue** [comment]
Also, needs to be `[BUGFIX release]` prefix I believe (since it is broken in current release branch)...

**Comment by ef4** [comment]
I don't think this implementation is correct. It will not update is the value goes from boolean to non-boolean. 

Basically it's always very suspicious to look at `value()` when setting up bindings -- it locks you into whatever behavior goes with the value you had at that moment.

**Comment by bekzod** [comment]
@ef4 cool, your solution looks more legit 👍, closing this one

**Pull request closed without merging**